store/postgres/src/deployment_store.rs

```diff
@@ -357,52 +357,111 @@ impl DeploymentStore {
-
-                    let _section = stopwatch.start_section("apply_entity_modifications_update");
-                    layout
-                        .update(conn, &key, data, block_number(ptr))
-                        .map(|_| 0)
+                    overwrites
+                        .entry(key.entity_type.clone())
+                        .or_insert_with(Vec::new)
+                        .push((key, data));
                 }
-                Insert { key, data } => {
-                    let section = stopwatch.start_section("check_interface_entity_uniqueness");
-                    self.check_interface_entity_uniqueness(conn, layout, &key)?;
-                    section.end();
-
-                    let _section = stopwatch.start_section("apply_entity_modifications_insert");
-                    layout
-                        .insert(conn, &key, data, block_number(ptr))
-                        .map(|_| 1)
+                Remove { key } => {
+                    removals
+                        .entry(key.entity_type.clone())
+                        .or_insert_with(Vec::new)
+                        .push(key);
                 }
-                Remove { key } => layout
-                    .delete(conn, &key, block_number(ptr))
-                    // This conversion is ok since n will only be 0 or 1
-                    .map(|n| -(n as i32))
-                    .map_err(|e| {
-                        anyhow!(
-                            "Failed to remove entity ({}, {}, {}): {}",
-                            key.subgraph_id,
-                            key.entity_type,
-                            key.entity_id,
-                            e
-                        )
-                        .into()
-                    }),
-            }?;
-            count += n;
+            }
+        }
+        // apply modification groups
+        for (entity_type, entities) in inserts.iter_mut() {
+            count += self.insert_entities(entity_type, entities, conn, layout, ptr, &stopwatch)?;
+        }
+        for (entity_type, entities) in overwrites.iter_mut() {
+            count +=
+                self.overwrite_entities(entity_type, entities, conn, layout, ptr, &stopwatch)?;
+        }
+        for (entity_type, entity_keys) in removals.into_iter() {
+            count +=
+                self.remove_entities(entity_type, entity_keys, conn, layout, ptr, &stopwatch)?;
         }
         Ok(count)
     }
 
+    fn insert_entities(
+        &self,
+        entity_type: &EntityType,
+        data: &mut Vec<(EntityKey, Entity)>,
+        conn: &PgConnection,
+        layout: &Layout,
+        ptr: &EthereumBlockPointer,
+        stopwatch: &StopwatchMetrics,
+    ) -> Result<i32, StoreError> {
+        let section = stopwatch.start_section("check_interface_entity_uniqueness");
+        for (key, _) in data.iter() {
+            // WARNING: This will potentially execute 2 queries for each entity key.
```

Comment:
Good point - we should figure out how to batch this check, too, but that can wait for another day. Maybe file an issue for just that?